x/interchainstaking/keeper/hooks.go

```diff
@@ -14,6 +16,33 @@ func (k Keeper) BeforeEpochStart(_ sdk.Context, _ string, _ int64) {
 }
 
 func (k Keeper) AfterEpochEnd(ctx sdk.Context, epochIdentifier string, epochNumber int64) {
+	// every day
+	if epochIdentifier == epochstypes.EpochIdentifierDay {
+
+		k.Logger(ctx).Info("handling day end", "epoch_identifier", epochIdentifier, "epoch_number", epochNumber)
+		k.Logger(ctx).Debug("flushing outstanding delegations for the day")
+		k.IterateZones(ctx, func(index int64, zone types.Zone) (stop bool) {
+			addressBytes, err := utils.AccAddressFromBech32(zone.DelegationAddress.Address, zone.AccountPrefix)
```

Comment:
wrap the below in `if zone.DelegationAddress != nil` as we cannot guarantee it will be non-nil when an epoch passes